core/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -680,17 +679,27 @@ object DeltaTable {
   }
 
   /**
-   * Check if the provided `identifier` string, in this case a file path,
-   * is the root of a Delta table using the given SparkSession.
+   * Check if the provided `identifier` string represents a Delta table.
+   * Identifier strings can be either a path or a table name.
+   * In the case of a file path, it checks if it is the root of a Delta
+   * table using the given SparkSession.
    *
-   * An example would be
+   * Examples are as follows:
    * {{{
    *   DeltaTable.isDeltaTable(spark, "path/to/table")
+   *   DeltaTable.isDeltaTable(spark, "MyDeltaTable")
    * }}}
    *
    * @since 0.4.0
    */
   def isDeltaTable(sparkSession: SparkSession, identifier: String): Boolean = {
+    // Try table name first
```

Comment:
So I think what we should do is basically have two results, `isDeltaTableName` & `isDeltaTablePath` and then return `isDeltaTableName || isDeltaTablePath` basically (don't necessarily have to do exactly that just the flow of the reasoning.) `isDeltaTableName` would be false if the `identifier` is an invalid table name.

The way it's currently implemented gives the wrong result in the scenario that I have a delta table in a `identifier` dir in my current directory. Providing it's relative path here will be false, but prior to this change it would be true.